### PR TITLE
fix wd

### PR DESCRIPTION
**Pull request by antinucleon**

### CODE REVIEW

**Comment by antinucleon** [comment]
@winstywang

File: python/mxnet/optimizer.py [hunk 1]
```diff
@@ -86,11 +86,23 @@ def _init_cc_optimizer(name, param_keys, param_vals):
...
         self.lr_scale = {}
         self.num_update = 0
         self._index_update_count = {}
+        self.specialized = False
+        self.weight_set = set([])
+        if arg_names is not None:
+            self.specialized = True
+            index = 0
+            for name in arg_names:
+                if name.endswith('data') or name.endswith('label'):
```
Comment (winstywang) [inline python/mxnet/optimizer.py]: We had better doc this convention?

**Comment by winstywang** [comment]
It should be good to merge this PR

**Comment by winstywang** [comment]
Also should we change model.py to pass in the arg_names by default?

**Comment by antinucleon** [comment]
Model.py is changed already. Seems there is compile problem in new mshadow or dmlc core. will check it in morning.

**Comment by winstywang** [comment]
Sorry, got it. Then no problem to merge.

**Pull request merged**